.github/workflows/handle_new_contribution.yml

```diff
@@ -0,0 +1,51 @@
+name: Handle New Contribution
+
+on:
+  pull_request_target:
+    types: [opened, ready_for_review]
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  assign_and_label:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.head.repo.fork == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+      - name: Print Context
+        run: |
+          echo "$GITHUB_CONTEXT"
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+      - uses: actions/cache@v3
+        with:
+          path: .venv
+          key: ${{ runner.os }}-poetry-${{ hashFiles('poetry.lock') }}
+      - name: Install Python Dependencies
+        run: |
+          poetry install
+      - name: Add an assignee and a Contribution label
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          PR_URL: ${{github.event.pull_request.html_url}}
+        run: |
+          echo "Getting the contribution TL"
+          cd Utils/github_workflow_scripts
+          poetry run python ./handle_new_contribution.py
+          contrib_tl=$( cat contrib_tl.txt )
+          echo "$contrib_tl"
+          find . -name "contrib_tl.txt" -type f -delete
+          echo "Finished retrieving the contribution TL username"
+          echo "Add an assignee and a Contribution label"
+          gh pr edit --add-assignee "$contrib_tl" --add-label Contribution "$PR_URL"
```

Comment:
isn't it possible to do all of this in the PR instead of writing a file and reading it after using the script?

take a look at the implementation in the `handle_external_pr.py` in content.